lib/jekyll-readme-index/generator.rb

```diff
@@ -67,7 +68,12 @@ def dir_has_index?(relative_path)
 
     # Regexp to match a file path against to detect if the given file is a README
     def readme_regex
-      @readme_regex ||= %r!/readme(#{Regexp.union(markdown_converter.extname_list)})$!i
+      regex = option(REGEX_KEY)
```

Comment:
This method is called for each file and it is probably not too smart to fetch the config each time. I am open to suggestions on how to improve this. Is the config available statically? If so, I could hoist this to a class-local variable/field.